packages/salesforcedx-utils-vscode/src/commands/channelService.ts

```diff
@@ -37,6 +38,9 @@ export class ChannelService {
   }
 
   public streamCommandStartStop(execution: CommandExecution) {
+    if (SfdxSettingsService.getEnableClearOutputBeforeEachCommand()) {
```

Comment:
The channel service is my first choice to attempt to handle this preference in one central location.  Unfortunately, the way the channel service is used throughout the app is very inconsistent.  This "streamCommandStartStop()" is used by some of the abstract executors ( SfdxCommandletExecutor, LibraryExectutor ) that are exported from salesforcedx-utils-vscode.